include/zephyr/net/socket.h

```diff
@@ -205,6 +207,12 @@ struct zsock_addrinfo {
 	char _ai_canonname[DNS_MAX_NAME_SIZE + 1];
 };
 
+struct tls_dtls_peer_cid {
```

Comment:
I wonder, to we even need to specify a custom data type for the option? In the end, we only expect to retrieve the CID and an information about the CID length. The `getsockopt()` could:
1. Verify the `optlen` parameter, i.e. whether it's not smaller than `MBEDTLS_SSL_CID_OUT_LEN_MAX`
2. Copy the CID directly into `optval` buffer provided, and set the `optlen` value to the actual length ofthe CID.
3. In case no peer CID is set during the handshake, just set the `optlen` to 0.

What do you think?

Reply:
Yes, that's simpler for the Zephyr side.  It made a sample I'm working on a bit simpler but that's not a big deal.

Reply:
Well, one thing we do lose is the ability to distinguish between a peer that does not support CID with one that chose to send an empty CID, indicating it supports CID but does not need the other side to send a particular value back.

We could return 0 from tls_opt_dtls_peer_connection_id_get() and set *optlen to -1 to signify the peer does not support it.  Or we could return -ENOTSUP from tls_opt_dtls_peer_connection_id_get().

Thoughts?

Reply:
Ok, I missed the fact that there is a difference between CID not set and CID not supported.

The option length is of type `socklen_t`, so we can't really set it to a negative value. I'm also not a fan of `getsockopt()` returning an error based on the option value. IMO In that case I think it's better to stick to the current solution.

Reply:
OK, I undid my local changes and am back to using the struct.